test/sql-tap/CMakeLists.txt

```diff
@@ -9,3 +9,124 @@ build_module(sql_uuid sql_uuid.c)
+          sql_uuid decimal
+)
+
+list(APPEND TEST_ENV
+  "LUA_PATH=${LUA_PATH}"
+  "LUA_CPATH=${LUA_CPATH}"
+  "BUILDDIR=${CMAKE_CURRENT_BINARY_DIR}"
+)
+
+file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *.test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  # FIXME: By default, GLOB lists directories.
+  # Directories are omitted in the result if LIST_DIRECTORIES
+  # is set to false. New in version CMake 3.3.
+  if(${test_name} STREQUAL ${TEST_SUITE_NAME})
+    continue()
+  endif()
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}${TEST_PARAM_DELIM}memtx")
+  set(work_dir "${CMAKE_CURRENT_BINARY_DIR}/${test_name}-workdir-memtx")
+  file(MAKE_DIRECTORY ${work_dir})
+  set(TEST_CMD "${RM} -rf ${work_dir}/* && ${TARANTOOL_BIN} ${test_path}")
+  add_test(NAME ${test_title}
+           COMMAND ${BASH} -c "${TEST_CMD}"
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+  set_tests_properties(${test_title} PROPERTIES
+    ENVIRONMENT "${TEST_ENV};TT_WORK_DIR=${work_dir};TEST_ENGINE=memtx;"
+    LABELS "${TEST_SUITE_NAME};integration"
+    DEPENDS ${TEST_SUITE_NAME}-deps
+    FAIL_REGULAR_EXPRESSION "${CTEST_FAIL_REGEXP}"
+  )
+
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}${TEST_PARAM_DELIM}vinyl")
+  set(work_dir "${CMAKE_CURRENT_BINARY_DIR}/${test_name}-workdir-vinyl")
+  file(MAKE_DIRECTORY ${work_dir})
+  set(TEST_CMD "${RM} -rf ${work_dir}/* && ${TARANTOOL_BIN} ${TAP_TEST_FLAGS} ${test_path}")
+  add_test(NAME ${test_title}
+           COMMAND ${BASH} -c "${TEST_CMD}"
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+  set_tests_properties(${test_title} PROPERTIES
+    ENVIRONMENT "${TEST_ENV};TT_WORK_DIR=${work_dir};TEST_ENGINE=vinyl;"
+    LABELS "${TEST_SUITE_NAME};integration"
+    DEPENDS ${TEST_SUITE_NAME}-deps
+    FAIL_REGULAR_EXPRESSION "${CTEST_FAIL_REGEXP}"
+  )
+endforeach()
+
+# See test/sql-tap/suite.ini.
+string(JOIN ";" DISABLED_TESTS
+  test/sql-tap/analyze1.test.lua
+  test/sql-tap/analyze3.test.lua
+  test/sql-tap/analyze4.test.lua
+  test/sql-tap/analyze5.test.lua
+  test/sql-tap/analyze6.test.lua
+  test/sql-tap/analyze7.test.lua
+  test/sql-tap/analyze8.test.lua
+  test/sql-tap/analyze9.test.lua
+  test/sql-tap/analyzeC.test.lua
+  test/sql-tap/analyzeD.test.lua
+  test/sql-tap/analyzeE.test.lua
+  test/sql-tap/analyzeF.test.lua
+  test/sql-tap/date.test.lua
+  # See gh-3694.
+  test/sql-tap/debug_mode_only.test.lua
+  test/sql-tap/e_expr.test.lua
+  test/sql-tap/gh-3350-skip-scan.test.lua
+  test/sql-tap/like2.test.lua
+  test/sql-tap/randexpr1.test.lua
+  test/sql-tap/selectA.test.lua
+  test/sql-tap/tkt3791.test.lua
+  test/sql-tap/tkt-bd484a090c.test.lua
+  test/sql-tap/types2.test.lua
+)
+foreach(test_name IN LISTS DISABLED_TESTS)
+  set_tests_properties(
+    ${test_name}${TEST_PARAM_DELIM}memtx
+    ${test_name}${TEST_PARAM_DELIM}vinyl
+    PROPERTIES DISABLED TRUE
+  )
+endforeach()
+
+# See test/sql-tap/suite.ini.
```

Comment:
Since #11043 is landed, should the part below be deleted?

Reply:
Many thanks!

```diff
@@ -116,17 +116,3 @@ foreach(test_name IN LISTS DISABLED_TESTS)
     PROPERTIES DISABLED TRUE
   )
 endforeach()
-
-# See test/sql-tap/suite.ini.
-string(JOIN ";" LONG_TESTS
-  test/sql-tap/gh-3332-tuple-format-leak.test.lua
-  test/sql-tap/gh-3083-ephemeral-unref-tuples.test.lua
-)
-foreach(test_name IN LISTS LONG_TESTS)
-  set_tests_properties(
-    ${test_name}${TEST_PARAM_DELIM}memtx
-    ${test_name}${TEST_PARAM_DELIM}vinyl
-    PROPERTIES
-      LABELS "long"
-  )
-endforeach()
```